src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,144 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationServiceImpl registrationServiceImpl;
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        registrationServiceImpl = new RegistrationServiceImpl();
+        user = new User();
+        Storage.people.clear();
+    }
+
+    @Test
+    void userIsValid_ok() {
+        User existingUser = new User();
+        existingUser.setLogin("differentLogin");
+        Storage.people.add(existingUser);
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validPas");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_ageLessThan18_notOk() {
+        user.setAge(17);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        user.setAge(-5);
+        user.setLogin("validLogin");
+        user.setPassword("validePasssword");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_ageEquals18_ok() {
+        user.setAge(18);
+        user.setLogin("valLog");
+        user.setPassword("valPas");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_LoginExist_notOk() {
+        User existingUser = new User();
+        existingUser.setLogin("validLogin");
+        Storage.people.add(existingUser);
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void registre_LoginLength_notOk() {
+        user.setAge(20);
+        user.setLogin("log");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
```

Comment:
The test method 'registre_LoginLength_notOk' has a typo in its name ('registre' instead of 'register') and does not specify what about the login length is not OK. Consider renaming it to 'register_shortLogin_notOk' to be more descriptive.